### PR TITLE
check for topic hierarchy on dataset publishing (#431)

### PR DESCRIPTION
**Pull request by tomkralidis**
This PR does a match/check of topic hierarchy between discovery metadata and data mappings at the time of design time workflow. Addresses #431.

### CODE REVIEW

File: wis2box-management/wis2box/metadata/discovery.py [hunk 1]
```diff
@@ -176,6 +177,10 @@ def publish(ctx, filepath, verbosity):
     try:
         dm = DiscoveryMetadata()
         record_mcf = dm.parse_record(filepath.read())
+
+        if record_mcf['wis2box']['topic_hierarchy'] not in DATADIR_DATA_MAPPINGS['data'].keys():  # noqa
```
Comment (webb-ben) [inline wis2box-management/wis2box/metadata/discovery.py:181]: fwiw `record_mcf['wis2box']['topic_hierarchy'] not in DATADIR_DATA_MAPPINGS['data'].keys()` == `record_mcf['wis2box']['topic_hierarchy'] not in DATADIR_DATA_MAPPINGS['data']`

File: wis2box-management/wis2box/data/__init__.py [hunk 2]
```diff
@@ -201,11 +203,10 @@ def add_collection(ctx, filepath, verbosity):
     """Add collection index to API backend"""
 
     meta = gcm(filepath.read())
-    # topic_hierarchy = meta['id']
 
-    # th, _ = validate_and_load(topic_hierarchy)
+    if meta['topic_hierarchy'] not in DATADIR_DATA_MAPPINGS['data'].keys():
+        raise click.ClickException('No matching topic hierarchy')
```
Comment (maaikelimper) [inline wis2box-management/wis2box/data/__init__.py:208]: exception = f"topic_hierarchy={meta['topic_hierarchy']} not found in data-mappings\n"
exception += f"topic_hierarchy defined in data-mappings: {DATADIR_DATA_MAPPINGS['data'].keys()}" 
raise click.ClickException(exception)

File: wis2box-management/wis2box/metadata/discovery.py [hunk 3]
```diff
@@ -176,6 +177,10 @@ def publish(ctx, filepath, verbosity):
...
+            raise click.ClickException('No matching topic hierarchy')
```
Comment (maaikelimper) [inline wis2box-management/wis2box/metadata/discovery.py:182]: exception = f"topic_hierarchy={meta['topic_hierarchy']} not found in data-mappings\n"
exception += f"topic_hierarchy defined in data-mappings: {DATADIR_DATA_MAPPINGS['data'].keys()}" 
raise click.ClickException(exception)

**Review by maaikelimper — COMMENTED** [review]
suggest to make the error a bit more verbose to help user resolve the issue

**Review by maaikelimper — APPROVED** [review]

**Pull request merged**